File: ExampleApp/app/src/main/java/com/motiv/example/dao/AuthTokenDao.java
package com.motiv.example.dao;

import androidx.fragment.app.*;
import androidx.room.*;
import com.motiv.example.AuthToken;
import dagger.*;
import dagger.android.*;
import dagger.android.support.*;
import javax.inject.*;

@Dao
public interface AuthTokenDao {

    @Insert
    public long saveAuthToken(com.motiv.example.AuthToken authToken);

    @Delete
    public int deleteAuthToken(com.motiv.example.AuthToken authToken);

    @Query("SELECT * FROM authToken")
    public java.util.List<AuthToken> getAuthTokens();

    @Query("SELECT * FROM authToken WHERE id = :id")
    public AuthToken getAuthToken(java.lang.String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void saveAuthTokens(java.util.List<com.motiv.example.AuthToken> authToken);
}

File: ExampleApp/app/src/main/java/com/motiv/example/dao/AuthTokenDao.kt
 
package com.motiv.example.dao
import androidx.room.*
import com.motiv.example.AuthToken
import kotlin.collections.List
@Dao
public interface AuthTokenDao {

    @Insert
    public fun saveAuthToken(authToken: com.motiv.example.AuthToken): Long@Delete
    public fun deleteAuthToken(authToken: com.motiv.example.AuthToken): Int@Query("SELECT * FROM authToken")
    public fun getAuthTokens(): List<AuthToken>@Query("SELECT * FROM authToken WHERE id = :id")
    public fun getAuthToken(id: String): AuthToken@Insert(onConflict = OnConflictStrategy.REPLACE)
    public fun saveAuthTokens(authToken: List<com.motiv.example.AuthToken>): Unit
}
